### DISCUSSION TITLE
Configuration Notification Delegate

### DISCUSSION THREAD

**Discussion by wopsononock** [pkp/pkp-lib#8582] [Proposals]
I don't see a way of assigning persistent, task-style notifications to users that allow us to pass along customized messages about things such as plugin misconfiguration or administrative tasks like managing other user accounts.  Example: pkp-plagiarism plugin has no username/password configured so submissions are not checked for cheating.  We have an error message from the plugin that details what went wrong.  Revealing it to the journal managers as a task gives them awareness of a problem and points them at the remedy.

I propose creating a new notification delegate for creating these types of tasks and have a draft implementation at https://github.com/pkp/pkp-lib/pull/8581

**Comment by asmecher** [comment]
Thanks, @wopsononock, I've added a review to your code! The whole notification system needs rewriting, but in the meantime, sticking with existing patterns to add a few new options is fine with me.
